src/figuresClasses.ts

```diff
@@ -1,19 +1,109 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea() : number;
+  a?: number;
+  b?: number;
+  c?: number;
+  radius?: number;
+  width?: number;
+  height?: number;
 }
 
-export class Triangle {
+enum Figures {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Figures.Triangle;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.shape = Figures.Triangle;
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid side length');
+    }
+
+    if (c >= (a + b)) {
```

Comment:
what if other sides will be incorrect? try to check all 3 sides to be smaller than other 2